pandas/core/frame.py

```diff
@@ -1810,50 +1810,59 @@ def info(self, verbose=None, buf=None, max_cols=None, memory_usage=None,
         lines.append(self.index.summary())
 
         if len(self.columns) == 0:
-            lines.append('Empty %s' % type(self).__name__)
+            lines.append('Empty {name}'.format(name=type(self).__name__))
             _put_lines(buf, lines)
             return
 
         cols = self.columns
+        cols_count = len(cols)
 
         # hack
         if max_cols is None:
-            max_cols = get_option('display.max_info_columns',
-                                  len(self.columns) + 1)
+            max_cols = get_option('display.max_info_columns', cols_count + 1)
 
         max_rows = get_option('display.max_info_rows', len(self) + 1)
 
         if null_counts is None:
-            show_counts = ((len(self.columns) <= max_cols) and
+            show_counts = ((cols_count <= max_cols) and
                            (len(self) < max_rows))
         else:
             show_counts = null_counts
-        exceeds_info_cols = len(self.columns) > max_cols
+        exceeds_info_cols = cols_count > max_cols
 
         def _verbose_repr():
-            lines.append('Data columns (total %d columns):' %
-                         len(self.columns))
-            space = max(len(pprint_thing(k)) for k in self.columns) + 4
+            lines.append('Data columns (total '
+                         '{count} columns):'.format(count=cols_count))
+            space = max((len(pprint_thing(k)) for k in cols))
+            space = max(space, len(pprint_thing('Column'))) + 4
+            space_num = len(pprint_thing(cols_count))
+            space_num = max(space_num, len(pprint_thing('Index'))) + 2
             counts = None
 
-            tmpl = "%s%s"
+            header = _put_str('Index', space_num) + _put_str('Column', space)
```

Comment:
Maybe call this 'N' rather than 'Index'. Also can you put a line of '-' after the header, formatted the same